src/hooks/renderTarget.ts

```diff
@@ -15,3 +16,22 @@ export const useRenderTarget = (width = 0, height = 0) => {
     [width, height, renderTargetSize]
   );
 };
+
+export const useRenderTargets = (
+  layer: LayerV2,
+  width?: number,
+  height?: number
+) => {
+  const { renderTargetSize } = useStore().uiStore;
+  return useMemo(() => {
+    const numRenderTargets = layer.maxConcurrentBlocks + 1;
+    return Array.from(
+      { length: numRenderTargets },
+      (_) =>
+        new WebGLRenderTarget(
```

Comment:
We potentially could get fancy here. I'm noticing that if the max concurrent blocks changes from 5 to 6 for instance, we would free from memory the old render targets and instantiate 6 new ones. Ideally we would reuse the first five. At least one way we could do this is with a useReducer hook to manage state more precisely. Probably a premature optimization at this point, but may be worth a note for us to handle later on.